documentation/src/main/asciidoc/topics/con_off_heap_memory.adoc

```diff
@@ -0,0 +1,66 @@
+[id='off_heap_memory']
+= Storing Data in Off Heap Memory
+{brandname} can store cache entries in JVM heap memory or off heap.
+Off heap storage lets you use unmanaged, native memory space for the JVM.
+
+Off heap storage gives you less overhead per entry than in JVM heap memory.
+In other words, off heap storage uses less memory than heap storage for the same amount of data.
+Another benefit of off heap storage is that it is not affected by Garbage Collector runs, which can improve overall JVM performance.
+Likewise you cannot change the JVM heap space at run-time but, when using off heap storage, {brandname} dynamically allocates memory.
+However there are some trade-offs such as that heap dumps become worthless.
+
+Consider the following illustration of memory space for a JVM process where
+{brandname} is running:
+
+image::offheap.svg[]
+
+JVM heap memory, as shown in the preceding diagram, holds Java objects and other application data.
+The heap is divided into young and old generations that help keep referenced objects in memory.
+The Garbage Collector (GC) process reclaims space from unreachable objects, running more frequently on the young generation memory pool.
+
+When {brandname} stores cache entries in JVM heap memory, GC runs can take longer to complete as you start adding data to your caches.
+Because GC is a fairly intensive process, longer and more frequent runs can degrade application performance.
+
+Using off heap memory configures {brandname} to store data in native memory, or available system memory that is outside JVM memory management.
+Depending on your JVM implementation, native memory is also allocated for thread stacks, Just-in-Time (JIT) compiler code, and so on.
+As an example, the preceding diagram shows two memory pools that are allocated from native memory:
+
+* Metaspace holds class metadata.
+* Code Cache holds generated native code and internal structures.
+
+Taking the diagram as an example, when you use off heap storage {brandname} adds data to the Code Cache memory pool.
```

Comment:
Technically they are allocated in the "Unknown" region as shown by native memory tracking.

Reply:
I've updated to keep the wording a little more general. "Infinispan allocates native memory to data" is probably the thing we should say rather than mentioning specific memory pools.